api/tests/integration/infrastructure/repositories/authentication-method-repository_test.js

```diff
@@ -881,6 +881,108 @@ describe('Integration | Repository | AuthenticationMethod', function () {
     });
   });
 
+  describe('#updateAuthenticationComplementByUserIdAndIdentityProvider', function () {
+    context('When authentication method exists for Pole Emploi', function () {
+      let authenticationMethod;
+      let clock;
+
+      beforeEach(function () {
+        clock = sinon.useFakeTimers(new Date('2020-01-02'));
+        const userId = databaseBuilder.factory.buildUser().id;
+        authenticationMethod = databaseBuilder.factory.buildAuthenticationMethod.withPoleEmploiAsIdentityProvider({
+          id: 123,
+          externalIdentifier: 'identifier',
+          accessToken: 'to_be_updated',
+          refreshToken: 'to_be_updated',
+          expiredDate: Date.now(),
+          userId,
+        });
+        return databaseBuilder.commit();
+      });
+
+      afterEach(function () {
+        clock.restore();
+      });
+
+      it('should update the authentication complement in database for pole emploi', async function () {
+        // given
+        const userId = authenticationMethod.userId;
+        const expiredDate = Date.now();
+        const authenticationComplement = new AuthenticationMethod.PoleEmploiAuthenticationComplement({
+          accessToken: 'new_access_token',
+          refreshToken: 'new_refresh_token',
+          expiredDate,
+        });
+
+        // when
+        await authenticationMethodRepository.updateAuthenticationComplementByUserIdAndIdentityProvider({
+          authenticationComplement,
+          userId,
+          identityProvider: 'POLE_EMPLOI',
+        });
+
+        // then
+        const [updatedAuthenticationComplement] = await knex('authentication-methods')
+          .pluck('authenticationComplement')
```

Comment:
**question**
C'est quoi `pluck` ? La doc knex ne m'aide pas vraiment à comprendre ... 

> This will pluck the specified column from each row in your results, yielding a promise which resolves to the array of values selected.

Reply:
j'avoue je l'ai copie de l'autre test 🙈 mais en anglais, pluck veut dire cueillir et d'apres la doc, il va retourner toutes les valeurs de la colonne, dans ce cas authentication complement, et la retourner dans un tableau